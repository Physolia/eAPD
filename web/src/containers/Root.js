import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import TagManager from 'react-gtm-module';
import { setLatestActivity } from '../actions/auth';

import App from './App';

// Set up Google Tag Manager
const tagManagerArgs = {
  gtmId: 'GTM-M56ZTV4'
};
TagManager.initialize(tagManagerArgs);

const Root = ({ history, store }) => {
  window.dataLayer.push({
    event: 'pageview'
  });

  // Create listener for location changing to track activity
  history.listen(() => {
    store.dispatch(setLatestActivity());
  });

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </Provider>
  );
};

Root.propTypes = {
  history: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  store: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default Root;
