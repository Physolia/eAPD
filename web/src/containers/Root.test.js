import { shallow } from 'enzyme';
import React from 'react';
import configureMockStore from 'redux-mock-store';

import Root from './Root';

const mockStore = configureMockStore([]);

describe('the Root component', () => {
  const props = { history: { listen: jest.fn() }, store: mockStore({}) };

  test('renders correctly', () => {
    const component = shallow(<Root {...props} />);
    expect(component).toMatchSnapshot();
  });
});
