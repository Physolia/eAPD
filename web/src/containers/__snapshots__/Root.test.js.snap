// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Root component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ConnectedRouterWithContext)
    history={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      }
    }
  >
    <App />
  </Connect(ConnectedRouterWithContext)>
</Provider>
`;
